File: frontend/src/types/character/ArmorySkills.ts
// Tripod 타입 정의
export interface Tripod {
  Tier: number; // 트라이포드 단계
  Slot: number; // 슬롯 번호
  Name: string; // 트라이포드 이름
  Icon: string; // 트라이포드 아이콘 URL
  Level: number; // 트라이포드 레벨
  IsSelected: boolean; // 선택 여부
  Tooltip: string; // 트라이포드에 대한 설명 (HTML 문자열)
}

export interface Rune {
  Name: string;
  Icon: string;
  Grade: string;
  Tooltip: string;
}

// ArmorySkill 타입 정의
export interface ArmorySkill {
  Name: string; // 스킬 이름
  Icon: string; // 스킬 아이콘 URL
  Level: number; // 스킬 레벨
  Type: string; // 스킬 타입 (예: 일반, 각성 등)
  SkillType: number; // 스킬 타입 코드 (숫자값)
  Tripods: Tripod[]; // 트라이포드 정보 배열
  Rune: Rune;
  Tooltip: string; // 스킬 상세 툴팁 (HTML 문자열)
}

File: backend/src/main/java/kr/elfaka/lostark/character/dto/ArmorySkillsDto/ArmorySkills.java
package kr.elfaka.lostark.character.dto.ArmorySkillsDto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import java.util.List;

@Data
public class ArmorySkills {

    @JsonProperty("Name")
    private String name;  // 스킬 이름

    @JsonProperty("Icon")
    private String icon;  // 스킬 아이콘 URL

    @JsonProperty("Level")
    private int level;  // 스킬 레벨

    @JsonProperty("Type")
    private String type;  // 스킬 타입 (예: 일반, 각성 등)

    @JsonProperty("SkillType")
    private int skillType;  // 스킬 타입 코드 (숫자값)

    @JsonProperty("Tripods")
    private List<Tripod> tripods;  // 트라이포드 정보 배열

    @JsonProperty("Rune")
    private Rune rune;  // 룬 정보

    @JsonProperty("Tooltip")
    private String tooltip;  // 스킬 상세 툴팁 (HTML 문자열)
}
